--- ./package/base-files/Makefile.orig	2015-11-30 11:23:32.027998990 -0800
+++ ./package/base-files/Makefile	2015-11-30 11:24:19.365973177 -0800
@@ -31,7 +31,6 @@
 define Package/base-files
   SECTION:=base
   CATEGORY:=Base system
-  DEPENDS:=+netifd +libc +procd +jsonfilter +SIGNED_PACKAGES:usign
   TITLE:=Base filesystem for OpenWrt
   URL:=http://openwrt.org/
   VERSION:=$(PKG_RELEASE)-$(REVISION)
@@ -156,7 +155,7 @@
 	mkdir -p $(1)/root
 	$(LN) /proc/mounts $(1)/etc/mtab
 	rm -f $(1)/var
-	$(LN) /tmp $(1)/var
+	#$(LN) /tmp $(1)/var
 	mkdir -p $(1)/etc
 	$(LN) /tmp/resolv.conf /tmp/fstab /tmp/TZ $(1)/etc/
 
