Index: systemd-218/src/libsystemd/sd-bus/test-bus-memfd.c
===================================================================
--- systemd-218.orig/src/libsystemd/sd-bus/test-bus-memfd.c
+++ systemd-218/src/libsystemd/sd-bus/test-bus-memfd.c
@@ -151,6 +151,7 @@ int main(int argc, char *argv[]) {
         /* check content */
         assert_se(memcmp(buf, "ll", 2) == 0);
 
+#ifndef __UCLIBC__
         /* writev it out*/
         iov[0].iov_base = (char *)"ABC";
         iov[0].iov_len = 3;
@@ -173,6 +174,7 @@ int main(int argc, char *argv[]) {
         assert_se(memcmp(bufv[0], "ABC", 3) == 0);
         assert_se(memcmp(bufv[1], "DEF", 3) == 0);
         assert_se(memcmp(bufv[2], "GHI", 3) == 0);
+#endif /* __UCLIBC__ */
 
         sd_memfd_free(m);
 
