Index: systemd-218/src/shared/missing.h
===================================================================
--- systemd-218.orig/src/shared/missing.h
+++ systemd-218/src/shared/missing.h
@@ -305,6 +305,8 @@ static inline int name_to_handle_at(int
 #ifndef HAVE_SECURE_GETENV
 #  ifdef HAVE___SECURE_GETENV
 #    define secure_getenv __secure_getenv
+#  elif defined __UCLIBC__
+#    define secure_getenv getenv
 #  else
 #    error "neither secure_getenv nor __secure_getenv are available"
 #  endif
