Index: systemd-211/src/shared/util.c
===================================================================
--- systemd-211.orig/src/shared/util.c
+++ systemd-211/src/shared/util.c
@@ -85,6 +85,8 @@
 #include "def.h"
 #include "missing.h"
 
+#include "config.h"
+
 int saved_argc = 0;
 char **saved_argv = NULL;
 
@@ -6235,7 +6237,12 @@ int mkostemp_safe(char *pattern, int fla
 
         u = umask(077);
 
+#ifndef __UCLIBC__
         fd = mkostemp(pattern, flags);
+#else
+        fd = mkstemp(path);
+        if (fd >= 0) fcntl(fd, F_SETFD, FD_CLOEXEC);
+#endif
         if (fd < 0)
                 return -errno;
 
